scripts/plot_trees.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Plots all trees from an MCMC run.
+
+Example Usage:
+poetry run python ../scripts/plot_trees.py
+
+Example Usage with arguments:
+    poetry run python scripts/plot_trees.py
+    --out_dir data/plots/tree/mark00
+    --data_fp data/mcmc/mark00/samples.json
+
+"""
+
+import argparse
+import logging
+
+
+from pathlib import Path
+from tqdm import tqdm
+
+import pyggdrasil.serialize as serialize
+import pyggdrasil.tree_inference as tree_inf
+import pyggdrasil.visualize as visualize
+
+
+def create_parser() -> argparse.Namespace:
+    """
+    Parser for required input user.
+
+    Returns:
+        args: argparse.Namespace
+    """
+
+    parser = argparse.ArgumentParser(description="Plot trees from MCMC sample.")
+
+    parser.add_argument(
+        "--out_dir",
+        required=True,
+        help="Output directory to save the tree plots.",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--data_fp",
+        required=True,
+        help="File path containing the mcmc samples.",
+        type=str,
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def main() -> None:
+    """
+    Main function for plotting trees from MCMC samples.
+
+    Returns:
+        None
+    """
+
+    args = create_parser()
+
+    # get the full path
+    out_dir = Path(args.out_dir)
+    # if out_dir does not exist, create it
+    if not out_dir.exists():
+        out_dir.mkdir(parents=True, exist_ok=True)
+
+    # Set up logging
+    fullpath_log = out_dir / "plot_trees.log"
```

Comment:
If `out_dir` is a `Path`, then `fullpath_log` is also a `Path` and manual casting below is not really needed :wink:

Reply:
Oh great. - overlooked this.